sql/src/main/java/io/crate/action/sql/DDLStatementDispatcher.java

```diff
@@ -101,6 +103,35 @@ public DDLStatementDispatcher(BlobIndices blobIndices,
         }
 
         @Override
+        public ListenableFuture<Long> visitOptimizeTableStatement(OptimizeTableAnalyzedStatement analysis, UUID jobId) {
+            if (analysis.indexNames().isEmpty()) {
+                return Futures.immediateFuture(null);
+            }
+            ForceMergeRequest request = new ForceMergeRequest(analysis.indexNames().toArray(
+                new String[analysis.indexNames().size()]));
+
+            // Pass parameters to ES request
+            Integer maxNumSegments = analysis.settings().getAsInt(OptimizeSettings.MAX_NUM_SEGMENTS.name(), null);
+            if (maxNumSegments != null) {
+                request.maxNumSegments(maxNumSegments);
+            }
+            Boolean onlyExpugeDeletes = analysis.settings().getAsBoolean(OptimizeSettings.ONLY_EXPUNGE_DELETES.name(), null);
+            if (onlyExpugeDeletes != null) {
+                request.onlyExpungeDeletes(onlyExpugeDeletes);
+            }
+            Boolean flush = analysis.settings().getAsBoolean(OptimizeSettings.FLUSH.name(), null);
+            if (flush != null) {
+                request.flush(flush);
+            }
+
+            request.indicesOptions(IndicesOptions.lenientExpandOpen());
+
+            FutureActionListener<ForceMergeResponse, Long> listener = new FutureActionListener<>(Functions.<Long>constant(null));
```

Comment:
I'd like to responde a row count, so pls count all successfully merged indices (btw. otherwise this function could be static)